packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -498,8 +498,14 @@ class HotRunner extends ResidentRunner {
 
   void _resetDirtyAssets() {
     for (final FlutterDevice? device in flutterDevices) {
-      device!.devFS!.assetPathsToEvict.clear();
-      device.devFS!.shaderPathsToEvict.clear();
+      final DevFS? devFS = device!.devFS;
```

Comment:
`flutterDevices` is `List<FlutterDevice>`, not `FlutterDevice?`
```dart
    for (final FlutterDevice device in flutterDevices) {
      final DevFS? devFS = device.devFS;
```

Reply:
done